File: app/src/main/java/com/ksjd/testem/AccountDetails.kt
package com.ksjd.testem

/** One card on the account. Timestamps are epoch seconds (0 = unknown). */
data class AccountDetails(
    val snr: String = "",
    val cardTypeName: String = "",
    val organizationName: String = "",
    val cardValidFrom: Long = 0,
    val cardValidTo: Long = 0,
    val ticketValidFrom: Long = 0,
    val ticketValidTo: Long = 0,
    val discountValidFrom: Long = 0,
    val discountValidTo: Long = 0,
    val creditLastBalance: Double? = null,
    val currencySymbol: String = "",
    val cardTemplateBase64: String = ""
)

data class AccountSnapshot(
    val userName: String,
    val cards: List<AccountDetails>
)

File: iosApp/iosApp/AccountDetails.swift
import Foundation

struct AccountDetails {
    var cardTypeName: String = ""
    var organizationName: String = ""
    var cardValidFrom: Int64 = 0
    var cardValidTo: Int64 = 0
    var ticketValidFrom: Int64 = 0
    var ticketValidTo: Int64 = 0
    var discountValidFrom: Int64 = 0
    var discountValidTo: Int64 = 0
    var creditLastBalance: Double?
    var currencySymbol: String = ""
    var cardTemplateBase64: String = ""
}
